app/lib/json_web_token.rb

```diff
@@ -0,0 +1,27 @@
+class JsonWebToken
+  SECRET_KEY = Rails.application.secrets.secret_key_base.to_s
+
+  # Expects a hash {user_id: id}
+  def self.encode(payload, exp = 24.hours.from_now)
+    payload[:exp] = exp.to_i
+    begin 
+      JWT.encode(payload, SECRET_KEY)
+    rescue JWT::EncodeError
```

Comment:
This is a custom error. I am not seeing it defined in your PR. If you raise this error, it'll throw another error "uninitialized constant" error. In my branch, check app/lib/exceptions